cmake/std/trilinosprhelpers/TrilinosPRConfigurationBase.py

```diff
@@ -0,0 +1,592 @@
+
+        try:
+            # Generate files using ATDM/TriBiTS Scripts
+            if enable_map_entry is None:
+                cmd = [os.path.join( self.arg_workspace_dir,
+                                    'Trilinos',
+                                    'commonTools',
+                                    'framework',
+                                    'get-changed-trilinos-packages.sh'),
+                       os.path.join('origin', self.args.targetBranch),
+                       'HEAD',
+                       'packageEnables.cmake',
+                       'package_subproject_list.cmake']
+
+                print("")
+                print("packageEnables Command: \n$ {}\n".format(" \\\n    ".join(cmd)))
+
+                if not dryrun:
+                    subprocess.check_call(cmd)
+                else:
+                    print("-\n- SKIPPED DUE TO DRYRUN\n-")
+            else:
+                # Use the values in the PACKAGE_ENABLES section of the .ini file
+                with open('packageEnables.cmake',  'w') as f_out:
+                    f_out.write(dedent('''\
+                        MACRO(PR_ENABLE_BOOL  VAR_NAME  VAR_VAL)
+                          MESSAGE("-- Setting ${VAR_NAME} = ${VAR_VAL}")
+                          SET(${VAR_NAME} ${VAR_VAL} CACHE BOOL "Set in $CMAKE_PACKAGE_ENABLES_OUT")
+                        ENDMACRO()
+                        '''))
+
+                    for entry in enable_map_entry.split(" "):
+                        f_out.write("PR_ENABLE_BOOL(Trilinos_ENABLE_{} ON)\n".format(entry))
+                    #    '''
+                    #    PR_ENABLE_BOOL(Trilinos_ENABLE_''' + enable_map_entry + ''' ON)
+                    #    '''))
+                with open ('package_subproject_list.cmake', 'w') as f_out:
+                    f_out.write(dedent('''\
+                        set(CTEST_LABELS_FOR_SUBPROJECTS ''' + enable_map_entry + ''')
+                        '''))
+
+            print("")
+            print("Enabled Packages:")
+            cmd = ['cmake', '-P', 'packageEnables.cmake']
+            cmake_rstring=None
+            if not dryrun:
+                cmake_rstring = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            else:
+                print("-\n- SKIPPED DUE TO DRYRUN\n-")
+                cmake_rstring = str.encode("")
+            cmake_rstring = cmake_rstring.decode('utf-8')
+            print(cmake_rstring)
+
+        except subprocess.CalledProcessError as cpe:
+            print('There was an issue generating packageEnables.cmake. '
+                  'The error code was: {}'.format(cpe.returncode))
+            raise cpe
+
+        return 0
+
+
+    def validate_branch_constraints(self):
+        """
+        Verify that the source branch is allowed.
+
+        For the `master` branch, we only allow the source branch to be
+        a protected branch named with the scheme `master_merge_YYYYMMDD_HHMMSS`
```

Comment:
Just a heads-up:  This might be a point of fragility we want to deal with at some point in the future.  Maybe you also want to validate the PR author is legit before trying to merge something into `master`.

Reply:
@jmgate branches that match the rule 'master_merge_*' are protected branches so that limits who can actually create one, this is how we'd prevent just anybody from creating a branch to try and merge into the master branch.